docs/howto/operator/kubernetes.rst

```diff
@@ -25,180 +25,90 @@ KubernetesPodOperator
 The :class:`~airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator` allows
 you to create and run Pods on a Kubernetes cluster.
 
+.. contents::
+  :depth: 1
+  :local:
+
+.. note::
+  If you use `Google Kubernetes Engine <https://cloud.google.com/kubernetes-engine/>`__, consider
+  using the
+  :ref:`GKEStartPodOperator <howto/operator:GKEStartPodOperator>` operator as it
+  simplifies the Kubernetes authorization process.
+
 How does this operator work?
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-TODO: include a definition of how the operator works, highlight dependency management gains, language agnosticism,
-and a quick overview of a k8s pod
-Go through the lifecycle of the pod in the context of a single task being executed on Airflow:
-start, monitor, end, delete (if specified). Mention where the pods can be run (in Airflow cluster if already
-hosted on k8s or a different one if kube config is available)
-
-Reference content from:
-- https://cloud.google.com/composer/docs/how-to/using/using-kubernetes-pod-operator
-- https://www.astronomer.io/docs/kubepodoperator/
-- https://medium.com/bluecore-engineering/were-all-using-airflow-wrong-and-how-to-fix-it-a56f14cb0753
-
-How to define Configurations (ConfigMaps and Secrets)?
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-TODO: define general best practices from here: https://kubernetes.io/docs/concepts/configuration/overview/
-Have examples with `airflow.kubernetes.secret.Secret` and `airflow.kubernetes.volume.Volume`
+The ``KubernetesPodOperator`` is able to natively launch a Kubernetes Pod to run an individual task -
```

Comment:
The ``KubernetesPodOperato`` uses the Kubernetes API to launch a pod in a Kubernetes cluster.  By supplying an image URL and a bash command, the operator uses the Kube Python Client...